crates/runtime/src/accelerated_table/mod.rs

```diff
@@ -409,21 +412,48 @@ impl Builder {
     }
 
     /// Build the accelerated table
+    #[allow(clippy::too_many_lines)]
     pub async fn build(self) -> AcceleratedTableBuilderResult<AcceleratedTable> {
         let on_complete_notification = Arc::new(Notify::new());
 
         let (acceleration_refresh_mode, refresh_trigger) = match self.refresh.mode {
             RefreshMode::Disabled => (refresh::AccelerationRefreshMode::Disabled, None),
             RefreshMode::Append => {
                 if self.refresh.time_column.is_none() {
-                    // Get the append stream
-                    let Some(append_stream) = self.append_stream else {
-                        return AppendStreamRequiredSnafu.fail();
-                    };
-                    (
-                        refresh::AccelerationRefreshMode::Changes(append_stream),
-                        None,
-                    )
+                    // Check if this is a Pepper table or if there's a primary key constraint
+                    #[cfg(feature = "pepper")]
+                    let is_pepper = self
```

Comment:
Not a huge fan of this explicit check for pepper. Ideally we would have some other way to communicate that this provider supports append mode without time column

Reply:
I now require a time column. Decided to scope that out.